integration/nwo/network.go

```diff
@@ -773,27 +776,44 @@ func (n *Network) Bootstrap() {
 
 	n.bootstrapIdemix()
 
-	sess, err = n.ConfigTxGen(commands.OutputBlock{
-		ChannelID:   n.SystemChannel.Name,
-		Profile:     n.SystemChannel.Profile,
-		ConfigPath:  n.RootDir,
-		OutputBlock: n.OutputBlockPath(n.SystemChannel.Name),
-	})
-	Expect(err).NotTo(HaveOccurred())
-	Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
-
-	for _, c := range n.Channels {
-		sess, err := n.ConfigTxGen(commands.CreateChannelTx{
-			ChannelID:             c.Name,
-			Profile:               c.Profile,
-			BaseProfile:           c.BaseProfile,
-			ConfigPath:            n.RootDir,
-			OutputCreateChannelTx: n.CreateChannelTxPath(c.Name),
+	if n.SystemChannel != nil { // TODO this entire block could be removed once we finish using the system channel
+		sess, err = n.ConfigTxGen(commands.OutputBlock{
+			ChannelID:   n.SystemChannel.Name,
+			Profile:     n.SystemChannel.Profile,
+			ConfigPath:  n.RootDir,
+			OutputBlock: n.OutputBlockPath(n.SystemChannel.Name),
 		})
 		Expect(err).NotTo(HaveOccurred())
 		Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
 	}
 
+	if n.SystemChannel != nil { // TODO this entire block could be removed once we finish using the system channel
+		for _, c := range n.Channels {
+			sess, err := n.ConfigTxGen(commands.CreateChannelTx{
+				ChannelID:             c.Name,
+				Profile:               c.Profile,
+				BaseProfile:           c.BaseProfile,
+				ConfigPath:            n.RootDir,
+				OutputCreateChannelTx: n.CreateChannelTxPath(c.Name),
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, n.EventuallyTimeout).Should(gexec.Exit(0))
+		}
+	}
+
+	if n.SystemChannel == nil {
```

Comment:
isn't it better to simply check if it's nil, in which case to do the below and add also concatenation of TLS certs, and return?

Reply:
I am not sure what you are suggesting.
This part of the code uses `configtxgen` to create the genesis blocks of application channels, instead of the channel creation transaction to the system-channel. When the tests use the system channel, creating a channel means loading a pre-made tx. When it is not used, creating a channel means loading a pre-made genesis block. Eventually, when the system channel is removed, we will not need to prepare channel creation transactions at all; but now we still have to support both, for different tests.

What do you mean by "also concatenation of TLS certs"?

Reply:
I just meant, to move lines 804-815 to line 789 keeping the `if n.SystemChannel == nil` and inside that `if`, also add `n.ConcatenateTLSCACertificates()` as in line 817 and then `return`. 

This allows you to remove the `if n.SystemChannel != nil` in lines 779 and 790.

Reply:
Ah... got it. 

I did what you suggested but instead kept the `if n.SystemChannel != nil`, and removed the `if n.SystemChannel == nil`. This way when we finish removing the system channel we simply remove the whole  `if n.SystemChannel != nil` block.